opm/autodiff/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -189,6 +190,53 @@ namespace Opm
         std::vector<int> current_controls_;
     };
 
+    static inline data::Wells report( const WellStateFullyImplicitBlackoil& state,
+                                      const Wells* wells,
+                                      double step_len ) {
+
+        using rt = data::Rates::opt;
+
+        std::map< std::string, data::Well > res;
+
+        const int nw = state.numWells();
+        const int np = state.numPhases();
+
+        for( auto w = 0; w < nw; ++w ) {
+
+            std::map< size_t, data::Completion > completions;
+            const auto* begin = wells->well_connpos + w;
+            const auto* end = wells->well_connpos + w + 1;
+            for( auto* i = begin; i != end; ++i ) {
+
+                const auto perfrate = state.perfPhaseRates().begin() + *i;
+                data::Rates perfrates;
+                perfrates.set( rt::wat, *(perfrate + 0 ) );
+                perfrates.set( rt::oil, *(perfrate + 1 ) );
+                perfrates.set( rt::gas, *(perfrate + 2 ) );
+                /* needs conversion to logical cartesian index */
+                const size_t globalindex = wells->well_cells[ *i ];
```

Comment:
Right, which for the purpose of this function is sufficient to add the parameter, as its caller won't be merged in a while (but might live as local patches).